Refactor updates to the IDR static website

sbesson opened the pull request:
Similarly to a change that was made for www.openmicroscopy.org, this removes the requirement to track the explicit version of the static website deployed at https://idr.openmicroscopy.org/about/. The ome.deploy_archive job is replaced by a simple Python script which polls hourly for new GitHub releases and deploys them if it finds them.

The environment version is stored as a source of truth under /srv/www/VERSION and copied as part of the deployment script.

This was a long standing improvement in my backlog especially in terms of released but it was primarily motivated by https://github.com/IDR/deployment/commit/36fc6ce570b6e2b7d8b71f19acf1136565159640 as the current workflow for updating the IDR static website requires 1- a tag + release on the source repo 2- the execution of an Ansible playbook and 3- a commit against this repo to update the variables and prevent regression. All of this feels unnecessary complex for changes that could be as simple as updating a typo. 

In terms of workflow, this also means that the only requirement is to push a tag to https://github.com/IDR/idr.openmicroscopy.org and the new release should be automatically deployed within an hour like for https://www.openmicroscopy.org/.

pwalczysko commented:
Thank you @sbesson . How to set up a testing env ?
Can I maybe push a tag to my fork and then... (???)

sbesson commented:
I can deploy the changes to `test114`. I put the script manually for testing but I'd like to re-run the playbook on the environment.

Once done, I was thinking we could use https://github.com/IDR/idr.openmicroscopy.org/pull/177 as a way to test this i.e. as once this fix is merged and tagged, it should be updated on `test114` within the hour.

pwalczysko commented:
> can deploy the changes to `test114`. I put the script manually for testing but I'd like to re-run the playbook on the environment.

Sounds great, thank you

sbesson commented:
On `test114`, tested the current state of the PR against `test114` by deleting the existing `/srv/www` on the proxy and executing the playbook

```
TASK [Gathering Facts] ********************************************************************************************************************************
ok: [test114-proxy]

TASK [Install Python 3] *******************************************************************************************************************************
ok: [test114-proxy]

TASK [Create static directory] ************************************************************************************************************************
--- before
+++ after
@@ -1,10 +1,10 @@
 {
     "path": "/srv/www",
     "secontext": [
-        "unconfined_u",
+        "system_u",
         "object_r",
-        "var_t",
+        "httpd_sys_content_t",
         "s0"
     ],
-    "state": "absent"
+    "state": "directory"
 }

changed: [test114-proxy]

TASK [Set website displayed version] ******************************************************************************************************************
--- before
+++ after: /Users/sbesson/.ansible/tmp/ansible-local-80879ohk63ms6/tmpw_e2rhdu
@@ -0,0 +1 @@
+test114
\ No newline at end of file

changed: [test114-proxy]

TASK [Install deployment script] **********************************************************************************************************************
ok: [test114-proxy]

TASK [Run deployment script] **************************************************************************************************************************
changed: [test114-proxy]

TASK [Install Cron daemon] ****************************************************************************************************************************
ok: [test114-proxy]

TASK [Add cron job updating the website] **************************************************************************************************************
ok: [test114-proxy]

PLAY RECAP ********************************************************************************************************************************************
test114-proxy              : ok=8    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

The latest release of the static website should be deployed and we should be ready to test the update workflow after creating a new release

dominikl commented:
Does that mean idr-testing website should now have the link to IBM instead of aspera (ie. include https://github.com/IDR/idr.openmicroscopy.org/pull/177 ) ? Because it doesn't.

sbesson commented:
Although the PR is merged, it will need a new GitHub release to be deployed. @pwalczysko do you want to try pushing a tag?

sbesson commented:
:+1: that's exactly why I wanted someone else than me to test the workflow. Try again?

pwalczysko commented:
> 👍 that's exactly why I wanted someone else than me to test the workflow. Granted the required permissions to the @IDR/ui team. Do you want to try again?

The release happened successfully at 13.33, https://github.com/IDR/idr.openmicroscopy.org/releases/tag/2023.01.30 but I do not see the change on `idr-testing` or idr.

sbesson commented:
```
[sbesson@test114-proxy www]$ ls -alh
total 12K
drwxr-xr-x.  4 root root     69 Jan 30 14:00 .
drwxr-xr-x.  3 root root     17 Jan 27 22:00 ..
drwxr-xr-x. 20 1001 centos 4.0K Jan 27 22:00 2023.01.27
drwxr-xr-x. 20 1001 centos 4.0K Jan 30 14:00 2023.01.30
lrwxrwxrwx.  1 root root     19 Jan 30 14:00 html -> /srv/www/2023.01.30
-rw-r--r--.  1 root root      7 Jan 27 22:00 VERSION
```

From https://idr-testing.openmicroscopy.org/about/download.html I see `Last updated: 2023-01-30` and the Aspera link is updated. Were you expecting different changes?

pwalczysko commented:
> Were you expecting different changes?

No, and sorry, I can see it now as updated too. Maybe I was looking too early, or (despite my best effort) did not manage to clear cache.
LGTM
